vars/publishToCrates.groovy

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/** Library to publish rust artifacts to crates.io registry
+@param Map args = [:] args A map of the following parameters
+@param args.repository <required> - GitHub repository (source code) for publishing the artifacts to crates.io
+@param args.tag <required> - Associated GitHub tag
+@param args.packageToPublish <optional> - By default, the package in the current working directory is selected. The packageToPublish flag can be used to choose a different package in the workspace.
+*/
+void call(Map args = [:]) {
+    parameterCheck(args.repository, args.tag)
+    String packageToPublish = args.packageToPublish ? "-p ${args.packageToPublish}" : ''
+
+    checkout([$class: 'GitSCM', userRemoteConfigs: [[url: "${args.repository}" ]], branches: [[name: "${args.tag}" ]]])
+
+    withCredentials([string(credentialsId: 'crates-api-token', variable: 'API_TOKEN')]) {
+        sh "cargo publish ${packageToPublish} --dry-run && cargo publish ${packageToPublish} --token ${API_TOKEN}"
+    }
+}
+
+void parameterCheck(String repository, String tag) {
+    if (!repository || !tag) {
```

Comment:
Yes if either is not provided it should error out.

Reply:
Yeah so I want to error out if there is a tag but no repository which is correct. 
I believe what you are saying is `if (repository && tag)` which is equal to `if (!repository || !tag)` that I have above

Reply:
Have offline discussion with Sayali now, understand her intention is to test if tag exist when there IS a repository. If there is NO repository, then dont even need to care.

Suggest adding some comments on this in case the context is missed later, or just use 2 nested if condition for more clarity.

Thanks.